R/db-impala.R

```diff
@@ -0,0 +1,44 @@
+#' @export
+db_desc.Impala <- function(x) {
+  info <- DBI::dbGetInfo(x)
+
+  host <- if (info$servername == "") "localhost" else info$servername
+  port <- if (info$port == "") "" else paste0(":", port)
+
+  paste0(
+    info$dbms.name, " ", info$db.version,
+    "[", info$username, "@", host, port,
+    "/", info$dbname , "]")
+
+}
+
+#' @export
+sql_translate_env.Impala <- function(con) {
+
+  sql_variant(
+    scalar = dbplyr::sql_translator(
+      .parent = dbplyr::base_scalar,
```

Comment:
Since these are so similar, I think it's probably a good idea to define a `base_odbc_scalar`, `base_odbc_agg` etc so we can reduce the amount of duplicated code across these functions.